### DISCUSSION TITLE
How to deploy hono app with vercel template (not a nextjs template), in vercel?

### DISCUSSION THREAD

**Discussion by igmtdev** [honojs/hono#2676] [Q&A]
how to deploy hono app in vercel?, using the vercel template not the next.js template, I want to separate my hono server for some reason, I'm trying to run the script from hono vercel template which is "start": "vercel dev" to start my hono app but I've got this error Error: Cannot use import statement outside a module

![image4](https://github.com/honojs/hono/assets/161505752/2cb655c0-35e8-441d-a542-4cb307164e02)
![image](https://github.com/honojs/hono/assets/161505752/48a76f19-f0fb-4f20-91fc-77952f9a453b)
![image2](https://github.com/honojs/hono/assets/161505752/73ffeb2a-5454-41e3-be15-092b07d77755)
![image3](https://github.com/honojs/hono/assets/161505752/fb749e4e-69f8-470e-b9d1-0bd64c8ea692)

**Comment by ghost** [comment]
It looks like you are trying to use ES modules (using import/export syntax) in a CommonJS environment.
The easiest solution is to add this setting to `package.json`:

```jsonc
{
  "type": "module",
  // rest...
}
```

This resolves the error because all `.js` and `.ts` files are handled as ES modules.

**Reply by igmtdev** [reply]
I figured out that when I removed the "edge runtime" that caused the error, the reason why I removed the "edge runtime" is to use "bcrypt"

**Reply by rezcoco** [reply]
> I figured out that when I removed the "edge runtime" that caused the error, the reason why I removed the "edge runtime" is to use "bcrypt"

in that case you should use `bcryptjs` instead

**Comment by prashantchothani** [comment]
If .js file is used instead of .ts , while deploying on Vercel the following warning comes in the Build log:

Warning: Node.js functions are compiled from ESM to CommonJS. If this is not intended, add "type": "module" to your package.json file.
